app/views/users/show.html.erb

```diff
@@ -15,11 +15,12 @@
   <h1><%= @user.name %></h1>
 
   <div class="clear"></div>
-
-  <% if @user.banned? %>
-    <span class="user-banned">This user is banned!</span>
+  <% if @ban_json %>
+    <span class="user-banned">This user is banned for "<%=@ban_json["reason"]%>"<%=" until #{@ban_json["expires"]}" unless @ban_json["expires"] == "forever"%></span>
+  <% elsif @user.banned? %>
```

Comment:
Server bans and website bans are two separate things, so just use two separate if blocks instead of if/elsif here.